apps/gidd/management/commands/update_idps_sadd_estimates.py

```diff
@@ -14,55 +14,78 @@ class Command(BaseCommand):
+        """
+        super().__init__()
+        self.iso3_to_country_name_map = self.load_country_name_map()
+        self.iso3_to_country_id_map = self.load_country_id_map()
+
+    def load_country_name_map(self):
+        """
+        Loads a mapping from ISO3 country codes to country names.
+        Returns:
+            dict: A dictionary mapping ISO3 codes to country names.
+        """
+        return {country.iso3: country.name for country in Country.objects.all()}
+
+    def load_country_id_map(self):
+        """
+        Loads a mapping from ISO3 country codes to country IDs.
+        Returns:
+            dict: A dictionary mapping ISO3 codes to country IDs.
+        """
+        return {country.iso3: country.id for country in Country.objects.all()}
+
     def add_arguments(self, parser):
-        parser.add_argument('csv_file')
+        parser.add_argument('csv_file_path', type=str)
 
     def handle(self, *args, **kwargs):
-        def _format_cause(cause_string):
-            if cause_string == Crisis.CRISIS_TYPE.CONFLICT.label:
-                return Crisis.CRISIS_TYPE.CONFLICT.value
-            elif cause_string == Crisis.CRISIS_TYPE.DISASTER.label:
-                return Crisis.CRISIS_TYPE.DISASTER.value
+        csv_file_path = kwargs['csv_file_path']
+        try:
+            self.process_file(csv_file_path)
+            logger.info("IDPS SADD estimates updated successfully.")
+        except FileNotFoundError:
+            logger.error(f"CSV file at {csv_file_path} not found.")
+        except Exception as e:
+            logger.error(f"Failed to update IDPS SADD estimates: {str(e)}")
 
-        def format_number(number):
-            return number.strip() or None
+    def process_file(self, file_path):
+        with open(file_path, 'r') as file:
+            reader = csv.DictReader(file)
+            objects_to_create = [self.create_estimate(obj) for obj in reader if self.validate_row(obj)]
+        with transaction.atomic():
+            IdpsSaddEstimate.objects.all().delete()
+            IdpsSaddEstimate.objects.bulk_create(objects_to_create)
 
-        csv_file = kwargs['csv_file']
+    def validate_row(self, row):
+        required_keys = ['ISO3', 'Year', 'Sex', 'Cause']
+        if not all(key in row for key in required_keys):
+            logger.error("Missing required data fields in CSV.")
+            return False
+        return True
 
-        iso3_to_country_id_map = {
-            item['iso3']: item['id'] for item in Country.objects.values('iso3', 'id')
-        }
-        iso3_to_country_name_map = {
-            item['iso3']: item['idmc_short_name'] for item in Country.objects.values('iso3', 'idmc_short_name')
-        }
+    def create_estimate(self, item):
+        return IdpsSaddEstimate(
+            iso3=item['ISO3'],
+            country_name=self.iso3_to_country_name_map.get(item['ISO3']),
+            country_id=self.iso3_to_country_id_map.get(item['ISO3']),
```

Comment:
Since both of the field is required, let use `[]`

```suggestion
            country_id=self.iso3_to_country[item['ISO3']]['id'],
            country_name=self.iso3_to_country[item['ISO3']]['name'],
```